pkg/ir/spec.go

```diff
@@ -53,36 +57,151 @@ type (
 
 	// ConfigDefs holds definitions that are part of the network architecture
 	ConfigDefs struct {
-		Subnets map[string]IP
+		VPCs map[ID]*VPCDetails
 
-		// Network interface name to IP address
-		NIFToIP map[string]IP
+		Subnets map[ID]*SubnetDetails
 
-		// Instance is a collection of NIFs
-		InstanceToNIFs map[string][]string
+		NIFs map[ID]*NifDetails
 
-		// VPEs have a single IP
-		VPEToIP map[string]IP
+		Instances map[ID]*InstanceDetails
 
-		// list of VPC's cidrs
-		AddressPrefixes []CIDR
+		VPEEndpoints map[ID]*VPEEndpointDetails
```

Comment:
I think it is problematic, mostly because there is a whole bunch of things that are VPESomething so the eye gets used to it. That said, VPEReservedIPs or VPESubnetIPs can also work.